sympy/core/mul.py

```diff
@@ -317,7 +317,11 @@ def flatten(cls, seq):
                         # the exponent is an integer
                         if e.is_Rational:
                             if e.is_Integer:
-                                coeff *= Pow(b, e)  # it is an unevaluated power
+                                c = Pow(b, e)
```

Comment:
We should find a way so that it isn't necessary to call `Pow(b, e)` here. This routine should be able to handle an unevaluated Pow so it shouldn't need to force evaluation like this.

Reply:
Wouldn't merely creating an Pow object with large enough exponential `e` remain unevaluated here. Since below that we check for `is_Number` and a multiplication with `c` which requires forced evaluation that's probably for smaller exponential ( bigger exponential like `Pow(200,300)` give `is_Number` as `False` and remain unevaluated while being appended to `c_part` ) . So I don't think there is a forced evaluation here at-least for bigger exponential.

( This is a part of original work from #19948 , so I may be wrong here. Correct me if so )

Reply:
Regardless of having a limit on the evaluation of large exponentials it should in general be possible to work with unevaluated exponentials without forcing them to evaluate. Part of the work involved in fixing this is removing this kind of forced evaluation from the codebase.

Reply:
Ah, I see. I'll make the necessary changes once I fix the current failing tests.